modules/user_access/main.tf

```diff
@@ -0,0 +1,121 @@
+# User and Roles
+
+locals {
+  role_users = [
+    for r in var.roles : {
+      name   = r.name
+      policy = r.policy
+      users = [
+        for u in var.users :
+        u.name if contains(u.roles, r.name)
+      ]
+    }
+  ]
+
+  role_user_arns = [
+    for r in local.role_users : {
+      name = r.name
+      user_arns = [
+        for u in aws_iam_user.access_user :
+        u.arn if contains(r.users, trimprefix(u.name, "${var.project}-"))
+      ]
+    }
+  ]
+}
+
+# Create users and groups with polcy for AWS resource access
+resource "aws_iam_group" "access_group" {
+  count = length(var.roles)
+  name  = var.roles[count.index].name
+  path  = "/users/"
+}
+
+resource "aws_iam_policy" "access_group" {
+  count       = length(var.roles)
+  name        = "${var.project}-${var.roles[count.index].name}-${var.environment}"
+  description = "Group policy"
+  policy      = var.roles[count.index].policy
+}
+
+resource "aws_iam_group_policy_attachment" "access_group" {
+  count      = length(var.roles)
+  group      = aws_iam_group.access_group[count.index].name
+  policy_arn = aws_iam_policy.access_group[count.index].arn
+}
+
+## Group policy
```

Comment:
Can you remove this if it's not necessary?

Reply:
done